alphadia/workflow/optimization.py

```diff
@@ -343,6 +344,9 @@ def _update_optimization_manager(self):
             {"fwhm_mobility": fwhm_mobility_at_optimum}
         )
 
+        batch_index_at_optimum = self.history_df["batch_idx"].loc[index_of_optimum]
```

Comment:
Maybe add this info from the description as a comment here
`take the batch index of the optimum, at the cost of potentially getting the batch library twice if these are the same. (But the time impact of this is negligible and the benefits can be significant.)...`